### PR TITLE
Add EIP: Hashtag NFT Collective Royalty Treasury

### PR DESCRIPTION
**Pull request by livetreetech**
When opening a pull request to submit a new EIP, please use the suggested template: https://github.com/ethereum/EIPs/blob/master/eip-template.md

We have a GitHub bot that automatically merges some PRs. It will merge yours immediately if certain criteria are met:

 - The PR edits only existing draft PRs.
 - The build passes.
 - Your GitHub username or email address is listed in the 'author' header of all affected PRs, inside <triangular brackets>.
 - If matching on email address, the email address is the one publicly listed on your GitHub profile.

### CODE REVIEW

File: EIPS/eip-royalty-treasury-proposal.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
+---
+eip: royalty-treasury-proposal
```
Comment (Pandapip1) [inline EIPS/eip-royalty-treasury-proposal.md:2]:
```suggestion
eip: 6145
```
Don't forget to rename the file!

Comment (abcoathup) [reply EIPS/eip-royalty-treasury-proposal.md:2]: @Pandapip1 should this be #6144 as this was the original PR?

Comment (livetreetech) [reply EIPS/eip-royalty-treasury-proposal.md:2]: No that md was corrupt upon commit, thus rejected. Please continue with 6145

**Comment by livetreetech** [comment]
@Pandapip1 the file was renamed from - to _ are we saying it is correct now or should be reverted back to - naming convention? (Thank you for the help!)

**Comment by livetreetech** [comment]
Happy holiday, please can someone advise on what the next steps are with this PR? It seem ready to me are there any issues outstanding?

**Comment by livetreetech** [comment]
> Pandapip1

Hope you had a good start to 2023. Is there anything I can do to progress this? I do not see what issues are outstanding with the PR?

File: EIPS/eip-royalty-treasury-proposal.md (hunk 1 above)
Comment (Pandapip1) [inline EIPS/eip-royalty-treasury-proposal.md:2]:
```suggestion
eip: 6145
```

**Review by livetreetech — COMMENTED** [review]
@Erik4569 @Pandapip1 @abcoathup

Thanks for the help with this, I cannot see why I am getting a Walidator error with a relative link to the license.md?  Any suggestions you might have would be much appreciated.  THANK YOU

Error: error[markdown-rel-links]: non-relative link or image
  --> EIPS/eip-[6](https://github.com/ethereum/EIPs/actions/runs/3884271693/jobs/6626612769#step:3:7)145.md
   |
[8](https://github.com/ethereum/EIPs/actions/runs/3884271693/jobs/6626612769#step:3:9)4 | Copyright and related rights waived via [CC0](../LICENSE.md).
   |
Error: validation found errors :(

**Comment by livetreetech** [comment]
@Erik4569 @Pandapip1 @abcoathup

Would appreciate any suggestions here, looks like a bug with the 
Walidator 
tried ./ and ../ now path to /License.md

?

**Comment by SamWilsn** [comment]
Sorry about the confusion with the markdown line numbers. `eipw` incorrectly reports external link errors as coming from the last line in the file.

File: EIPS/eip-6145.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
+---
+eip: 6145
+title: Hashtag NFT Collective Royalty Treasury
+description: A method to deposit and retrieve royalties from an Ethereum Address
```
Comment (SamWilsn) [inline EIPS/eip-6145.md:4]:
```suggestion
description: A method to deposit and retrieve royalties from an Ethereum address
```

File: EIPS/eip-6145.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+discussions-to: https://ethereum-magicians.org/t/eip-6145-proposal-hashtag-nft-collective-royalty-treasury/12152
+status: Draft
+type: Standards Track
+category: ERC
+created: 2022-12-15
+---
+
+## Abstract
+
+This standard allows social media networks, NFT marketplaces and other websites to deposit royalties associated with an Ethereum address. The Ethereum address may be associated with a human readable name (e.g. a hashtag), to pay-out royalties and retrieve royalty payment information. The standard is loosely coupled and generic to maximise the potential use cases.  Unlike other EIP proposals [EIP-2981](./eip-2718.md) or [EIP-4910](./eip-4910.md) which embody mechanisms for calculating royalties, this standard sets out the mechanism for depositing and paying out royalties. The goal is to increase the earning potential for creators rather than defining the business logic for royalty payments.
```
Comment (SamWilsn) [inline EIPS/eip-6145.md:15]: > Unlike other EIP proposals [EIP-2981](./eip-2718.md) or [EIP-4910](./eip-4910.md) which embody mechanisms for calculating royalties, this standard sets out the mechanism for depositing and paying out royalties.

This bit should be in motivation, not abstract. The abstract should just be a terse technical summary.

File: EIPS/eip-6145.md [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+Further implementation details may be found in the Livetree repository found at the livetreetech/LivetreeCollective github repo.
+
+## Security Considerations
+
+Important for security discussions, surfaces risks and could be the use of a whitelist described in the HashtagNFTCollectiveResolver DepositRoyalty method.
+
+## Copyright
+
+Copyright and related rights waived via [CC0](/LICENSE.md).
```
Comment (SamWilsn) [inline EIPS/eip-6145.md:84]:
```suggestion
Copyright and related rights waived via [CC0](../LICENSE.md).
```

File: EIPS/eip-6145.md [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+|   |                              | Unique    | 0xA50e98f9cb301c04B9BB34d1BD95c2Dc8F3e8Ff3 |
+|   |                              | Quartz    | 0xA50e98f9cb301c04B9BB34d1BD95c2Dc8F3e8Ff3 |
+| 2 | HashtagNFTCollectiveResolver | Ethereum  | 0xA50e98f9cb301c04B9BB34d1BD95c2Dc8F3e8Ff3 |
+|   |                              | Moonbeam  | 0x9a6Cba29cc0cA4f18990F32De15aae08819F4cD0 |
+|   |                              | Moonriver | 0x9a6Cba29cc0cA4f18990F32De15aae08819F4cD0 |
+|   |                              | Celo      | 0xdEc0F21665065e0c67b8eD107b1767c95cC4A763 |
+|   |                              | Unique    | 0x9a6Cba29cc0cA4f18990F32De15aae08819F4cD0 |
+|   |                              | Quartz    | 0x9a6Cba29cc0cA4f18990F32De15aae08819F4cD0 |
+
+Further implementation details may be found in the Livetree repository found at the livetreetech/LivetreeCollective github repo.
```
Comment (SamWilsn) [inline EIPS/eip-6145.md:76]: Please remove any external links. Yes, we count this as an external link ;)

File: EIPS/eip-6145.md [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+| 2 | GetHashtagNFTBalances _(view)_ | **Inputs** _address tokenAddress_ - An Ethereum address **Outputs** uint256 - Returns sum of the royalties for given Ethereum address | Facilitates querying an Ethereum address and returning the royalties associated to the Ethereum address. The method **MUST** be called with an Ethereum address. |
+| 3 | WithdrawFromTreasuryToAddress | **Inputs** _address tokenAddress_ - Address royalty was associated in the deposit _address receiver_ - Royalty payout Ethereum address _uint256 amount_ - Royalty amount to be withdrawn to the receiver **Outputs** none emits **Withdraw** event _Withdraw(senderAddr, receiver, amount, tokenAddr);_ | Withdraws the given amount of the royalty (blockchain native token) associated with the Ethereum address to the receiver. The receiver **MAY** be a wallet, smart contract or other Ethereum address.  The tokenAddress **MUST** be an address the sender has deposited associated tokenAddress. The implementation of this method **SHOULD** restrict the caller and amount of withdrawal through the use of a whitelist defined in DepositRoyalty.
+
+### HashtagNFTCollectiveResolver
```
Comment (SamWilsn) [inline EIPS/eip-6145.md:49]: If these headings are code snippets, you should put them in backticks:

```suggestion
### `HashtagNFTCollectiveResolver`
```

File: EIPS/eip-6145.md [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
...- Array of Ethereum addresses _uint256[] calldata tokenRoyalties_ - Matching array of royalty deposit amounts for each Ethereum Address  **Outputs** none  emits **Deposit** event _event Deposit(senderAddr, address[] tokenAddresses, uint256[] tokenRoyalties);_ | This method optimises the gas costs relating to the deposit of royalties for a number of different associated Ethereum addresses. It takes an array of tokenAddresses (Ethereum addresses) associated with an array of tokenRoyalties (Royalty amounts).  tokenAddresses array **MUST** be paired equally and **MUST** be in the same order as tokenRoyalties. The deposit payable amount sent to this function **MUST** be in the blockchain’s native token (for example: ETH, GLMR, MOVR, CELO, UNQ etc). The payable amount of the native token **MUST** equal the sum of all tokenRoyalties. A human readable name registered in the naming service **MAY** be associated with the Ethereum address(es) provided in tokenAddresses. The implementation of this method stores the sender address in a “whitelist” along with the quantity of the blockchain’s native token deposited which is in turn is used to permission the method WithdrawFromTreasuryToAddress. |
```
Comment (SamWilsn) [inline EIPS/eip-6145.md:45]: I see a couple "MUST"s in here. If you're setting out requirements, they have to go in the Specification section.

The Backwards Compatibility section should just be for describing breaking changes. For example, if you were extending EIP-721 NFTs into soulbound tokens, you'd note here that soulbound tokens can't be transferred (unlike regular 721 NFTs).

File: EIPS/eip-6145.md [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+
+The #Hashtag NFT Collective Royalty Treasury standard comes about as a result of many years of work to help realise the overarching ambition of Web 3.0 decentralisation, namely, to empower people to collectively address some of the world’s most challenging problems such as climate change, global inequality to domestic abuse.  Hash-tagging (#) media has become a defacto standard across Web 2.0 social networks for not only garnering support (e.g. #BLM, etc). The introduction of a decentralised hashtag standard aims to harness the power of social media in a decentralised fashion to bring about the change Web 3.0 promises, fundamentally giving creators the potential to earn income (“royalties”) from content they post and are tagged from a variety of social networks. The first implementation of this standard can be downloaded via www.livetree.com. The generic design facilitates additional royalties to be attributed to NFTs or any smart contract or wallet address.
+
+## Specification
+
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
+
```
Comment (SamWilsn) [inline EIPS/eip-6145.md:25]: Your standard isn't standardizing anything. The meat of your proposal should be here.

**Comment by YaroslavPysanskyi** [comment]
@YaroslavPysanskyi

**Comment by yamommasofat** [comment]
I'm here. Had to attend to my hammock.

File: EIPS/eip-6145.md (hunk 4 above)
Comment (yamommasofat) [reply EIPS/eip-6145.md:84]: 0x00217211201ea57f90429dd387bbde31d0e4d2f21143de3189470121480f3094

**Review by yamommasofat — APPROVED** [review]
https://github.com/ethereum/EIPs/actions/runs/5236504390/jobs/9454085491

**Review by yamommasofat — APPROVED** [review]

**Comment by yamommasofat** [comment]
How do I use this information to get my stake off of this catacomb nightmare of ruckus.

**Comment by yamommasofat** [comment]
I'm so sick of this crap. I want to take my eth and leave. I have followed all direction. I need money. I haven't had any in 6 months. Where is my fucking exit

**Comment by SamWilsn** [comment]
@yamommasofat while I appreciate your plight, this is not the correct forum for discussing it. This repository contains technical standards for Ethereum developers. You might have better luck on [/r/ethereum](https://www.reddit.com/r/ethereum/) or attending an AllCoreDevs call to voice your concerns.

image

I also ask that you keep the discussion here civil.

**Comment by livetreetech** [comment]
> This pull request was closed due to inactivity. If you are still pursuing it, feel free to reopen it and respond to any feedback or request a review in a comment.

There has been no meaningful feedback on the PR - can someone from Ethereum foundation please comment or let me know why it wasn't passed? I only see some bizarre cartoon pictures that seem unrelated

**Pull request closed without merging**